packages/runtime/src/runtime/hooks/useNavigateModal.tsx

```diff
@@ -50,40 +65,29 @@ export const useNavigateModalScreen: EnsembleActionHook<
     }
   }, [isStringAction, action]);
 
-  useEffect(() => {
-    if (!action || !modalContext || isComplete !== false) {
-      return;
-    }
+  const navigateCommand = useCommandCallback(
+    (evalContext, ...args) => {
+      if (!action || !modalContext) return;
 
-    navigateModalScreen(
-      action,
-      modalContext,
-      applicationContext?.application,
-      evaluatedInputs.inputs,
-      title,
-      onDismissCallback,
-    );
-    setIsComplete(true);
-  }, [
-    action,
-    applicationContext?.application,
-    evaluatedInputs.inputs,
-    isComplete,
-    modalContext,
-    onDismissCallback,
-    title,
-  ]);
+      const context = merge({}, evalContext, args[0]);
 
-  return useMemo(() => {
-    const callback = (args: unknown): void => {
-      if (!action) {
-        return;
-      }
+      const evaluatedInputs =
+        !isString(action) && !isNil(action.inputs)
+          ? evaluateInputs(cloneDeep(action.inputs), screenModel, context)
+          : {};
 
-      setIsComplete(false);
-      setContext(args);
-    };
+      navigateModalScreen(
+        action,
+        modalContext,
+        applicationContext?.application,
+        evaluatedInputs,
+        title,
+        onDismissCallback,
```

Comment:
We can get rid of the extra hook and just pass the ensembleAction callback directly.